src/index.ts

```diff
@@ -1,38 +1,58 @@
-import {AmbiguousError} from './types/errors/ambiguous.js'
-import {NotFound} from './types/errors/not-found.js'
-import {AddOnAttachmentWithConfigVarsAndPlan, AddOnWithRelatedData, Link} from './types/pg/data-api.js'
+import {APIClient} from '@heroku-cli/command'
+
+import {AmbiguousError} from './errors/ambiguous.js'
+import {NotFound} from './errors/not-found.js'
+import {AddOnWithRelatedData, ExtendedAddonAttachment, Link} from './types/pg/data-api.js'
 import {ConnectionDetails, ConnectionDetailsWithAttachment, TunnelConfig} from './types/pg/tunnel.js'
-import {getDatabase} from './utils/pg/databases.js'
+import DatabaseResolver from './utils/pg/databases.js'
 import getHost from './utils/pg/host.js'
-import {exec} from './utils/pg/psql.js'
+import PsqlService from './utils/pg/psql.js'
 import {confirm} from './ux/confirm.js'
 import {prompt} from './ux/prompt.js'
 import {styledHeader} from './ux/styled-header.js'
 import {styledJSON} from './ux/styled-json.js'
 import {styledObject} from './ux/styled-object.js'
 import {table} from './ux/table.js'
 import {wait} from './ux/wait.js'
+
 export const types = {
-  errors: {
-    AmbiguousError,
-    NotFound,
-  },
   pg: {
-    AddOnAttachmentWithConfigVarsAndPlan: {} as AddOnAttachmentWithConfigVarsAndPlan,
     AddOnWithRelatedData: {} as AddOnWithRelatedData,
     ConnectionDetails: {} as ConnectionDetails,
     ConnectionDetailsWithAttachment: {} as ConnectionDetailsWithAttachment,
+    ExtendedAddonAttachment: {} as ExtendedAddonAttachment,
     Link: {} as Link,
     TunnelConfig: {} as TunnelConfig,
   },
 }
 
 export const utils = {
+  errors: {
+    AmbiguousError,
+    NotFound, // This should be NotFoundError for consistency, but we're keeping it for backwards compatibility
+  },
   pg: {
-    databases: getDatabase,
+    fetcher: {
```

Comment:
Do we want to expose this under the `fetcher` namespace? I know that's how it is organized currently. I don't have a strong opinion either way, but especially now that database is organized as a class, `fetcher` doesn't seem necessaty? Like, now you're instantiating a DatabaseResolver thing.